src/NLog/Targets/Wrappers/AutoFlushTargetWrapper.cs

```diff
@@ -68,11 +68,18 @@ public class AutoFlushTargetWrapper : WrapperTargetBase
         /// Delay the flush until the LogEvent has been confirmed as written
         /// </summary>
         /// <docgen category='General Options' order='10' />
-        public bool AsyncFlush { get => _asyncFlush ?? true;
+        public bool AsyncFlush
+        {
+            get => _asyncFlush ?? true;
             set => _asyncFlush = value;
         }
         private bool? _asyncFlush;
 
+        /// <summary>
+        /// Ignore explicit flush on configuration reload or application shutdown
+        /// </summary>
+        public bool IgnoreExplicitFlush { get; set; }
```

Comment:
Not sure about this name. I think "explicit" is also a bit difficult here (I think it should be implicit then)

Proposal: FlushOnEvents.

Or, It would be cool if we could split into 2 options, FlushOnShutdown and FlushOnReload, but I expect that's far more difficult to implement?

Reply:
Not possible to split into more settings. It will ignore config-reload, application-shutdown and explicit-flush. All of them call the explicit `LogManager.Flush()`

Reply:
What about, OnlyFlushOnCondition? 

(PS: LogManager.Flush() is explicit, Logmanager.ShutDown is implicit IMO)

Reply:
To me everything is explicit, except the implicit flush performed by autoflushwrapper.

Reply:
Renamed property to `FlushOnConditionOnly`